src/main/java/org/spongepowered/api/plugin/PluginContainer.java

```diff
@@ -52,10 +57,19 @@
     String getVersion();
 
     /**
-     * Returns the created instance of {@link Plugin}.
+     * Returns the assigned logger to this {@link Plugin}.
+     *
+     * @return The assigned logger
+     */
+    default Logger getLogger() {
+        return LoggerFactory.getLogger(getId());
+    }
+
+    /**
+     * Returns the created instance of {@link Plugin} if it is available.
      *
-     * @return The instance
+     * @return The instance if available
      */
-    Object getInstance();
+    Optional<Object> getInstance();
```

Comment:
Again, Forge will not give you a container without an instance in it if it could not create the mod object. Meaning calling ``` getPluginManager().getPlugin(someId) ``` will ALWAYS return either an empty Optional or one containing a container with a mod instance inside except for one exception (I think, needs testing).

Exception being the construct event. At that point, FML has the list of mod candidates and creates the containers. After construct, all the containers have instances (or are removed if creating the mod object fails). I'm not 100% sure on the latter.

Should I be correct about the above, this means that, for a very short window, it could come back at null on the instance. I'm of the strong opinion that the API shouldn't be made "silly" just because the Forge side doesn't get its act together for a short time.